plugins/modules/site_workflow_manager.py

```diff
@@ -797,13 +838,63 @@ def get_site_name(self, site):
                 self.set_operation_result("failed", False, self.msg, "ERROR")
                 return None
 
-            site_name = '/'.join([parent_name, name])
-            self.log("Constructed site name: {}".format(site_name), "INFO")
-            return site_name
+            site_name_hierarchy = '/'.join([parent_name, name])
+            self.log("Constructed site name: {}".format(site_name_hierarchy), "INFO")
+            return site_name_hierarchy
+
+        except Exception as e:
+            error_message = "An error occurred while getting site name: {}".format(str(e))
+            self.log(error_message, "ERROR")
+            return None
+
+    def get_bulk_site_names(self, site, bulk_operation=True):
+        """
+        Get and return a list of constructed site names for areas, buildings, and floors.
+
+        Parameters:
+        - self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+
+        Returns:
+        - list: A list of constructed site names.
+
+        Description:
+        This method collects and constructs site names from areas, buildings, and floors
+        based on the Cisco DNAC version.
+        """
+        name_list = []
+        try:
+            for area in self.config[0].get('site', {}).get('area', []):
+                area_name = area.get('name')
+                area_parent_name_hierarchy = area.get('parent_name_hierarchy')
+                if area_name and area_parent_name_hierarchy:
+                    name_list.append("{}/{}".format(area_parent_name_hierarchy, area_name))
+                elif not area_parent_name_hierarchy:
+                    self.log("Missing parent name hierarchy for area: {}".format(area_name), "ERROR")
+
```

Comment:
self.log("Processing buildings for site names.", "DEBUG")

Reply:
Added this line